awx/api/views.py

```diff
@@ -2710,15 +2710,13 @@ def update_raw_data(self, data):
                 data['extra_vars'] = extra_vars
             ask_for_vars_dict = obj._ask_for_vars_dict()
             ask_for_vars_dict.pop('extra_vars')
-            if get_request_version(self.request) == 1:  # TODO: remove in 3.3
-                ask_for_vars_dict.pop('extra_credentials')
             for field in ask_for_vars_dict:
                 if not ask_for_vars_dict[field]:
                     data.pop(field, None)
-                elif field == 'inventory' or field == 'credential':
+                elif field == 'inventory':
                     data[field] = getattrd(obj, "%s.%s" % (field, 'id'), None)
-                elif field == 'extra_credentials':
-                    data[field] = [cred.id for cred in obj.extra_credentials.all()]
+                elif field == 'credentials':
+                    data[field] = [cred.id for cred in obj.credentials.all()]
```

Comment:
`data[field] = list(obj.credentials.values_list('id', flat=True))`?

Reply:
@cchurch I fear that could be less efficient. We hit `obj.credentials.all()` a _lot_, and it may, in fact, be prefetched (if it's not by now, it should be). That means that the credential list is already in-memory. The id list, on the other hand, is a new query.

Reply:
Good point, depends on how `obj` is loaded -- *IF* `obj` is loaded from a queryset with `prefetch_related('credentials')` then `obj.credentials.all()` won't result in a new query each time.